js/index.ts

```diff
@@ -0,0 +1,241 @@
+  }, []);
+
+  return netInfo;
+}
+
+/**
+ * Deprecated method to remove the listener. You should upgrade to the new API.
+ *
+ * @deprecated
+ *
+ * @param type The event type.
+ * @param handler The event listener.
+ */
+export function removeEventListener(
+  type: string,
+  handler: DeprecatedTypes.ChangeHandler,
+): void {
+  DeprecatedUtils.warnOnce();
+
+  if (type === DEPRECATED_CHANGE_EVENT_NAME) {
+    DeprecatedSubscriptions.remove(handler);
+  }
+}
+
+/**
+ * Deprecated method to get the current state. You should upgrade to the new `fetch` method and
+ * handle the new state type.
+ *
+ * @deprecated
+ */
+export function getConnectionInfo(): Promise<DeprecatedTypes.NetInfoData> {
+  DeprecatedUtils.warnOnce();
+  return NativeInterface.getCurrentState().then(DeprecatedUtils.convertState);
+}
+
+/**
+ * Deprecated method to tell if the current connection is "expensive". Only available on iOS. You
+ * should now call the `fetch` method and look at the `details.isConnectionExpensive` property.
+ *
+ * @deprecated
+ */
+export function isConnectionExpensive(): Promise<boolean> {
+  DeprecatedUtils.warnOnce();
+  return NativeInterface.getCurrentState().then(
+    DeprecatedUtils.isConnectionExpensive,
+  );
+}
+
+export const isConnected = {
+  /**
+   * Deprecated method to listen for changes to the connected boolean. You should now use the
+   * normal `addEventListener` method and look at the `isConnected` property.
+   *
+   * @deprecated
+   */
+  addEventListener: (
+    eventName: string,
+    handler: DeprecatedTypes.IsConnectedHandler,
+  ): DeprecatedTypes.Subscription => {
+    if (eventName !== DEPRECATED_CHANGE_EVENT_NAME) {
+      return {remove: (): void => {}};
+    }
+
+    const listener = (state: Types.NetInfoState): void => {
+      handler(DeprecatedUtils.isConnected(state));
+    };
+
+    _isConnectedListeners.set(handler, listener);
+    Subscriptions.add(listener);
+
+    return {
+      remove: (): void => {
+        Subscriptions.remove(listener);
+      },
+    };
+  },
+
+  /**
+   * Deprecated method to stop listening for changes to the connected boolean. You should now use
+   * the normal `addEventListener` method and look at the `isConnected` property.
+   *
+   * @deprecated
+   */
+  removeEventListener: (
+    _eventName: string,
+    handler: DeprecatedTypes.IsConnectedHandler,
+  ): void => {
+    const listener = _isConnectedListeners.get(handler);
+    listener && Subscriptions.remove(listener);
+    _isConnectedListeners.delete(handler);
+  },
+
+  /**
+   * Deprecated method to get the current is connected boolean. You should now use the normal
+   * `fetch` method and look at the `isConnected` property.
+   *
+   * @deprecated
+   */
+  fetch: (): Promise<boolean> => {
+    return NativeInterface.getCurrentState().then(DeprecatedUtils.isConnected);
+  },
+};
+
+export * from './internal/types';
+export * from './internal/deprecatedTypes';
+
+export const StateType = {
```

Comment:
This and CellularGeneration seem to be duplicating the internal/types.ts information - that seems sub-optimal. It looks like you're re-exporting internal/types above, so they should be available? One is a const and one is a type, maybe both should collapse to an enum or something. I'm pretty new at Typescript but I'd be surprised if list maintenance needs to be done in two locations

Reply:
Same as the comment before. You can't use type information string constants as normal JS string constants in Typescript. It's a bit annoying, but a small amount of duplication isn't too much of an issue.

I've moved these into the `type.ts` file so they at least live together. If someone know a way to avoid the duplication, that would be amazing!